### PR TITLE
Enable parallel unit testing

### PR DESCRIPTION
**Pull request by Martin-Molinero**
#### Description

- Enable parallel unit testing, Travis build now takes ~13min!
- Adding a few unit test and micro performance improvements
#### Related Issue

N/A
#### Motivation and Context

Faster tests are better tests
#### Requires Documentation Change

N/A
#### How Has This Been Tested?

Unit and regression test, many times
#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [x] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Messaging/StreamingApi.cs [hunk 1]
```diff
@@ -35,10 +33,10 @@ public static class StreamingApi
         public static readonly bool IsEnabled = Config.GetBool("send-via-api");
 
         // Client for sending asynchronous requests.
-        private static readonly RestClient Client = new RestClient("http://streaming.quantconnect.com")
+        private static readonly Lazy<RestClient> Client = new Lazy<RestClient>(() => new RestClient("http://streaming.quantconnect.com")
```
Comment (jaredbroad) [inline Messaging/StreamingApi.cs:36]: I'd prefer to delete rather than update this ancient code; which since its a new larger problem suggest we just revert these changes and fix this class properly in a dedicated PR.

File: Engine/Results/BaseResultsHandler.cs [hunk 2]
```diff
@@ -316,7 +314,7 @@ public virtual string SaveLogs(string id, List<LogEntry> logs)
         /// <param name="result">The results to save</param>
         public virtual void SaveResults(string name, Result result)
         {
-            File.WriteAllText(GetResultsPath(name), JsonConvert.SerializeObject(result, Formatting.Indented));
```
Comment (jaredbroad) [inline Engine/Results/BaseResultsHandler.cs:319]: I don't see the harm in leaving this indented for ease of use locally? Why did you choose to remove?

Comment (Martin-Molinero) [reply Engine/Results/BaseResultsHandler.cs:319]: So regression tests are as fast as possible, I'll revert the change here and only apply it for the `RegressionResultHandler` 👍

**Comment by jaredbroad** [comment]
🎉 tests #FTW !
